### PR TITLE
Update gitIgnore & dependencies

### PR DESCRIPTION
**Pull request by Goooler**
- TARGET_SDK_VERSION  to 30
- COMPILE_SDK_VERSION to 30
- CONSTRAINT_LAYOUT to 2.0.4
- CORE_KTX to1.3.2
- KTLINT to 0.40.0
- AGP to 4.1.1
- KOTLIN to 1.4.20
- gradle to 6.7.1

### CODE REVIEW

**Comment by cortinico** [comment]
Thanks for your contribution @Goooler 

> * KOTLIN to 1.4.20
> * gradle to 6.7.1

I'm actually holding on on Kotlin 1.4.20 for a couple of days more till Gradle ships 6.8, given that the embedded Kotlin version has been bumped to 1.4+ there.

6.8 was at RC1 a couple of weeks ago: https://github.com/gradle/gradle/releases/tag/v6.8.0-RC1 so stable should be out soon.

**Comment by Goooler** [comment]
@cortinico I also update .gitIgnore file and gradle.properties file, could you review it ?

**Review by cortinico — COMMENTED** [review]
Thanks for the contrib 👍 

- The changes to the `.gitignore` looks good to me. 
- The bump to Gradle 6.7.1 should probably be done in a separate PR (and it will require to regenerate the wrapper)
- Please amend those changes in Dependencies.kt: Kotlin bump and KtLint bump
- I'm afraid some of the config you enable in `gradle.properties` might end up having unexpected behavior to developers' build. There are some that are ok to enabled by default (e.g. `org.gradle.configureondemand=true`). I'd rather offer the others as commented out, so users can enable them if they wish.

**Comment by cortinico** [comment]
> @cortinico I also update .gitIgnore file and gradle.properties file, could you review it ?

Done, sorry for the late review 🙏

**Comment by Goooler** [comment]
- `kotlin-android-extensions` have been deprecated in Kotlin 1.4.20, use ViewBinding taking replace
- You mentioned have been commented out

**Comment by cortinico** [comment]
> Please amend those changes in Dependencies.kt: Kotlin bump and KtLint bump

This is still pending

Plus please rebase on top of `master` as I've bumped Gradle to 6.7.1. here https://github.com/cortinico/kotlin-android-template/pull/25?

File: app/src/main/java/com/ncorti/kotlin/template/app/MainActivity.kt [hunk 1]
```diff
@@ -3,24 +3,25 @@ package com.ncorti.kotlin.template.app
...
 import androidx.appcompat.app.AppCompatActivity
+import com.ncorti.kotlin.template.app.databinding.ActivityMainBinding
 import com.ncorti.kotlin.template.library.FactorialCalculator
 import com.ncorti.kotlin.template.library.android.NotificationUtil
-import kotlinx.android.synthetic.main.activity_main.*
 
 class MainActivity : AppCompatActivity() {
 
     private val notificationUtil: NotificationUtil by lazy { NotificationUtil(this) }
+    private val binding by lazy { ActivityMainBinding.inflate(layoutInflater) }
```
Comment (cortinico) [inline app/src/main/java/com/ncorti/kotlin/template/app/MainActivity.kt:13]: Please don't use `by lazy` but rather a `lateinit` as suggested by Google's official documentation: https://developer.android.com/topic/libraries/view-binding#kotlin

File: gradle.properties [hunk 2]
```diff
@@ -1,21 +1,13 @@
...
+android.useAndroidX=true
+#kapt.include.compile.classpath=false
+#kapt.incremental.apt=true
+#kapt.use.worker.api=true
+#kapt.verbose=true
 kotlin.code.style=official
+org.gradle.caching=true
+#org.gradle.configureondemand=true
+org.gradle.jvmargs=-Xmx1536m -Dfile.encoding=UTF-8
+org.gradle.parallel=true
```
Comment (cortinico) [inline gradle.properties:13]: Can you amend all the changes in this file? I'd rather review this change in a separate PR.

**Review by cortinico — APPROVED** [review]
Sorry for the late review but I was 🎅 -ing.
Added a couple of comments more.

Other than the changes in `gradle.properties` it's good to go 👍

File: app/src/main/java/com/ncorti/kotlin/template/app/MainActivity.kt (hunk 1 above)
Comment (Goooler) [reply app/src/main/java/com/ncorti/kotlin/template/app/MainActivity.kt]: Have changed

File: gradle.properties (hunk 2 above)
Comment (Goooler) [reply gradle.properties]: Revert `gradle.properties` to your version

**Review by cortinico — APPROVED** [review]

**Comment by cortinico** [comment]
Thank you very much for the contribution @Goooler 🙏

**Comment by Goooler** [comment]
> Thank you very much for the contribution @Goooler 🙏

Thanks for your review

**Pull request merged**